contracts/TokenBalanceOracle.sol

```diff
@@ -7,36 +7,61 @@ import "@aragon/os/contracts/lib/token/ERC20.sol";
 
 contract TokenBalanceOracle is AragonApp, IACLOracle {
 
-    bytes32 public constant CHANGE_TOKEN_ROLE = keccak256("CHANGE_TOKEN_ROLE");
+    bytes32 public constant SET_TOKEN_ROLE = keccak256("SET_TOKEN_ROLE");
+    bytes32 public constant SET_BALANCE_ROLE = keccak256("SET_BALANCE_ROLE");
+
     string private constant ERROR_TOKEN_NOT_CONTRACT = "ORACLE_TOKEN_NOT_CONTRACT";
 
     ERC20 public token;
+    uint256 public minBalance;
 
-    event ChangeToken(address _newToken);
+    event TokenSet(address token);
+    event MinimumBalanceSet(uint256 minBalance);
 
-    function initialize(address _token) external onlyInit {
-        initialized();
+    function initialize(address _token, uint256 _minBalance) external onlyInit {
+        require(isContract(_token), ERROR_TOKEN_NOT_CONTRACT);
 
         token = ERC20(_token);
+        minBalance = _minBalance;
+
+        initialized();
     }
 
     /**
-    * @notice Change token to `_token`
+    * @notice Update token address to `_token`
     * @param _token The new token address
     */
-    function changeToken(address _token) external auth(CHANGE_TOKEN_ROLE) {
+    function setToken(address _token) external auth(SET_TOKEN_ROLE) {
         require(isContract(_token), ERROR_TOKEN_NOT_CONTRACT);
         token = ERC20(_token);
 
-        emit ChangeToken(_token);
+        emit TokenSet(_token);
+    }
+
+    /**
+    * @notice Update minimum balance to `_minBalance`
+    * @param _minBalance The new minimum balance
+    */
+    function setMinBalance(uint256 _minBalance) external auth(SET_BALANCE_ROLE) {
+        minBalance = _minBalance;
+
+        emit MinimumBalanceSet(_minBalance);
     }
 
     /**
     * @notice ACLOracle
-    * @dev IACLOracle interface conformance
+    * @dev IACLOracle interface conformance.  If the ACLOracle permissioned function should specify the minimum balance,
+    *      it should be used with the modifier 'authP(SOME_ACL_ROLE, arr(minBalance))'
     */
     function canPerform(address _sender, address, bytes32, uint256[] how) external view returns (bool) {
-        uint256 balance = token.balanceOf(_sender); 
-        return how.length > 0 ? balance >= how[0] : balance > 0;
+
+        uint256 requiredBalance = how.length > 0 ? how[0] : minBalance;
```

Comment:
Consider calling this `minBalanceLocal` or something. Otherwise it's changing definitions. `requiredBalance` to me suggests it should be exactly that amount.